### PR TITLE
Add help screen layout

### PR DESCRIPTION
**Pull request by lubej**
## Solution

Add help screen on mobile version

- add `SwiperJs` dependency for carousel effect;

## Links

- https://www.loom.com/share/7b28e34d33144ee4a7a56402ec3b8ce6

### CODE REVIEW

**Comment by buberdds** [comment]
that lib throws some warnings `https://github.com/mui/material-ui/issues/33274` but I didn't inspect it yet

File: src/app/pages/HomePage/ParaTimeSelector/para-time-selector-utils.ts [hunk 1]
```diff
@@ -20,6 +20,19 @@ export abstract class ParaTimeSelectorUtils {
     }
   }
 
+  static showMobileHelpScreen(step: ParaTimeSelectorStep, isMobile: boolean) {
+    if (!isMobile) {
+      return false
+    }
+
+    switch (step) {
```
Comment (buberdds) [inline src/app/pages/HomePage/ParaTimeSelector/para-time-selector-utils.ts:28]: why are we using switch statement to handle boolean ? Can't we just write 
`step === ParaTimeSelectorStep.ShowHelpScreen`. Additionally, do we even need utils for such simple case?

**Comment by lubej** [comment]
> https://github.com/mui/material-ui/issues/33274

Good point. Will migrate to [SwiperJs](https://swiperjs.com/element). As `react-swipeable-views` seems unmaintained.

File: src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                  marginBottom: 3,
+                }}
+                src={step.icon}
+                alt={step.label}
+              />
+            ) : null}
+          </div>
+        ))}
+      </SwipeableViews>
+      <Typography component="h4" color={COLORS.white} sx={{ marginBottom: 5 }}>
```
Comment (buberdds) [inline src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx:92]: wrong font size

File: src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        {currentStep.label}
+      </Typography>
+      {activeStep < 2 && (
+        <MobileStepper
+          variant="dots"
+          steps={totalSteps}
+          position="static"
+          activeStep={activeStep}
+          backButton={<></>}
```
Comment (buberdds) [inline src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx:101]: usually we return null when we don't want to render component

File: src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    setActiveStep(index as AvailableSteps)
+  }
+
+  const onGetStartedClick = () => {
+    setParaTimeStep(ParaTimeSelectorStep.Explore)
+  }
+
+  return (
+    <HelpScreenContainer>
+      <SwipeableViews index={activeStep} onChangeIndex={onChangeSwipeableIndex}>
```
Comment (buberdds) [inline src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx:72]: Swipe area is very small so I usually end up with moving graph that is below tutorial

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx [hunk 5]
```diff
@@ -15,15 +15,16 @@ import QuickPinchZoom, { make3dTransformValue, UpdateAction } from 'react-quick-
...
-interface ParaTimeSelectorProps {
+interface ParaTimeSelectorBaseProps {
   disabled: boolean
 }
 
 const ParaTimeSelectorGlow = styled(Box, {
   shouldForwardProp: (prop: string) =>
-    !(['disabled'] as (keyof ParaTimeSelectorProps)[]).includes(prop as keyof ParaTimeSelectorProps),
-})<ParaTimeSelectorProps>(({ disabled, theme }) => ({
+    !(['disabled'] as (keyof ParaTimeSelectorBaseProps)[]).includes(prop as keyof ParaTimeSelectorBaseProps),
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/HomePage/ParaTimeSelector/index.tsx:26]: I rather leave it, for cases where `[disabled]` style selector is globally applied, to not conflict with styling.

File: src/app/pages/HomePage/ParaTimeSelector/para-time-selector-utils.ts (hunk 1 above)
Comment (lubej) [reply src/app/pages/HomePage/ParaTimeSelector/para-time-selector-utils.ts:28]: I thought that mobile version would be more complex. Simplified the simple `switch` statements to `if`s, but I would keep the utils. If you feel strongly against I can move it into component.

File: src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  const [activeStep, setActiveStep] = useState<AvailableSteps>(0)
+  const allSteps = useConstant(() => steps(t))
+  const totalSteps = allSteps.length
+  const currentStep = allSteps[activeStep]
+
+  const swiperElRef = useRef<HTMLElement>(null)
+
+  useEffect(() => {
+    const handleSlideChange = (e: Event) => {
+      const [swiper] = (e as SlideChangeEvent).detail
```
Comment (buberdds) [inline src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx:84]: Are these types correct? handleSlideChange e param is a type of SlideChangeEvent not Event, right?

File: src/index.tsx [hunk 7]
```diff
@@ -15,6 +15,10 @@ import './locales/i18n'
 Axios.defaults.baseURL = process.env.REACT_APP_API
 const queryClient = new QueryClient({})
 
+import { register } from 'swiper/element/swiper-element'
+// register Swiper custom elements
+register()
```
Comment (lukaw3d) [inline src/index.tsx:20]: How come we aren't using 'swiper/react'?

Comment (lukaw3d) [reply src/index.tsx:20]: ah, i see, "Swiper React components will likely to be removed in future versions. It is recommended to migrate to Swiper Element instead."

File: src/types/swiper.d.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'swiper-container': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
+      'swiper-slide': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
```
Comment (lukaw3d) [inline src/types/swiper.d.ts:5]: Waw their typescript support sucks.

You can extend global to improve event types too
```ts
  interface ElementEventMap {
    'slidechange': SlideChangeEvent
  }
```

File: src/index.tsx (hunk 7 above)
Comment (lukaw3d) [inline src/index.tsx:20]: I'd rename to registerSwiperElements so comment becomes obsolete

File: src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+          backButton={null}
+          nextButton={null}
+        />
+      )}
+      {activeStep > 1 && (
+        <Button variant="contained" color="primary" onClick={onGetStartedClick}>
+          {t('home.helpScreen.getStarted')}
+        </Button>
+      )}
+      ss
```
Comment (lukaw3d) [inline src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx:131]: typo, or something

**Review by buberdds — APPROVED** [review]

**Comment by lukaw3d** [comment]
Something broke on small screens with this PR:
| Before | After |
| --- | --- |
|  localhost_1234_ | localhost_1234_ (1) |

**Comment by lukaw3d** [comment]
I'm also seeing: 
image

File: src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+      <SwiperBox>
+        <swiper-container ref={swiperElRef} slides-per-view="1">
+          {allSteps.map((step, index) => (
+            <swiper-slide key={step.label}>
+              {Math.abs(activeStep - index) < totalSteps ? (
+                <Box
+                  component="img"
+                  sx={{
+                    display: 'block',
+                    width: '100%',
```
Comment (buberdds) [inline src/app/pages/HomePage/ParaTimeSelector/HelpScreen/index.tsx:96]: This is causing issue @lukaw3d was showing

**Pull request merged**